### DISCUSSION TITLE
Real time live ASR?

### DISCUSSION THREAD

**Discussion by SinanAkkoyun** [openai/whisper#925] [Q&A]
Hello, I am curious on how to implement ASR chunking for realtime transcription (with interim results so to speak). I am aware of VAD implementations, but I wanted to ask if chunking with overlaps might somehow be possible.

Thank you so much for helping me out!

**Comment by Beenyaa** [comment]
Hey, I've tried implementing ASR chunking with overlaps myself to get those interim results in real-time-ish transcription. The good news is, it's definitely possible! But, it isn't always the greatest.

I created a sliding window to make this work. Essentially, I concatenated the current recording with a portion of the previous recording, like this: `concatenated_recording = np.concatenate([previous_recording[-ovelap_length:], current_recording])`. Then, I transcribed the concatenated numpy array and used the results to prompt the next chunk.

Unfortunately, this method often leads to premature endings of sentences, which in my case can completely change the meaning of the speech being transcribed. If you manage to make this work better, I'd love to hear about it.

**Reply by Beenyaa** [reply]
Another thought, you might want to checkout https://github.com/ggerganov/whisper.cpp/tree/master/examples/stream#sliding-window-mode-with-vad if openai/whisper doesn't work for you as you expected it to. This is where I got the sliding window idea from and this is the project I am looking to use going forward until someone makes a break through with an openai/whisper implementaion of real-time transcription.

**Comment by SinanAkkoyun** [comment]
Hey, thank you for your reply! That is a nice approach, thank you for sharing!

**Reply by makaveli10** [reply]
@SinanAkkoyun [WhisperLive](https://github.com/collabora/WhisperLive) is real time and uses faster-whisper backend.

**Comment by Gldkslfmsd** [comment]
https://github.com/openai/whisper/discussions/1978 -- Whisper-Streaming has a real-time implementation and supports faster-whisper backend.
